### PR TITLE
SCP-3452 chain-index e2e tests

### PR DESCRIPTION
**Pull request by james-iohk**
Rest API tests that hit each chain-index endpoint. Designed to run on testnet, should be easy enough to extend for mainnet too if desired. Can look at extending this for more complex scenarios such as involving live transactions to check behaviour and performance.

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [ ] ~Formatting, materialized Nix files, PNG optimization, etc. are updated~
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

**Comment by james-iohk** [comment]
Will open after refactor of api requests

**Review by raduom — APPROVED** [review]
LGTM

File: plutus-chain-index/e2e-tests/sanity-test/spec/from-hash/stake-validator.spec.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import * as env from '../../testnet-env.json';
+import { expect } from "chai";
+const request = require("supertest")(env.url);
+
+//there's any example of stake validator onchain yet
+describe("POST /from-hash/stake-validator", () => {
+    let response, resBody;
+    before(async () => {
+        response = await request
+            .post("/from-hash/minting-policy")
```
Comment (koslambrou) [inline plutus-chain-index/e2e-tests/sanity-test/spec/from-hash/stake-validator.spec.ts:10]: should be `stake` here.

File: plutus-chain-index/e2e-tests/sanity-test/spec/from-hash/stake-validator.spec.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import * as env from '../../testnet-env.json';
+import { expect } from "chai";
+const request = require("supertest")(env.url);
+
+//there's any example of stake validator onchain yet
```
Comment (koslambrou) [inline plutus-chain-index/e2e-tests/sanity-test/spec/from-hash/stake-validator.spec.ts:5]: Add a TODO saying that this should be update once there's a stake validator on chain. 

I would comment the test as it's a duplication of another one.

**Review by sjoerdvisscher — APPROVED** [review]

File: plutus-chain-index/e2e-tests/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+## Sanity API tests
+
+### Pre-requisites:
+- You have a testnet instance of chain-index running somewhere
+- Chain-index is synced to at least `min_slot` in the environment file `sanity-tests/testnet-env.json`
+- Update URL in environment file if needed
+
+### Run tests:
+- `nvm use`
```
Comment (koslambrou) [inline plutus-chain-index/e2e-tests/README.md:11]: BTW, I think this is optional for users of `nix-shell` as `node` is already available (currently v14.17.4).

Comment (james-iohk) [reply plutus-chain-index/e2e-tests/README.md:11]: Sure. I could remove this cmd and nvmrc file and we always assume run in nix-shell..?

Comment (koslambrou) [reply plutus-chain-index/e2e-tests/README.md:11]: Yeah exactly! However, you can leave this for those people not using `nix-shell`, and say that it's unnecessary for those using `nix-shell`

**Review by koslambrou — APPROVED** [review]

**Pull request merged**